src/pg_client.py

```diff
@@ -57,7 +60,38 @@ def get_solver_rewards(self, start_block: str, end_block: str) -> DataFrame:
                 self.exec_query(query=batch_reward_query_barn, engine=engine)
             )
 
-        return pd.concat(results)
+        results_df = pd.concat(results)
+
+        # warn and merge in case of solvers in both environments
+        if not results_df["solver"].is_unique:
+            duplicated_entries = results_df[results_df["solver"].duplicated(keep=False)]
+            log.warning(
+                f"Solvers found in both environments:\n {duplicated_entries}.\n Merging results."
+            )
+
+            def merge_lists(series: Series) -> list | None:
+                merged = []
+                for lst in series:
+                    if lst is not None:
+                        merged.extend(lst)
```

Comment:
Should we do a check here to see if the merge adds duplicates? It might make debugging easier to figure out where the duplicates are introduced.

Reply:
I tried to come up with a clean solution but failed. So if you have a good idea, feel free to propose it.

In principle, we could check the final dataframe after merging. That is probably easiest.

If we want to check if while merging, we probably only want it for the addresses and not for the values. If we already have two very similar but different functions, we can probably just modify the two columns in function. Then we might as well just remove duplicates right away. But the code looks significantly more indirect then, with aggregating multiple columns into multiple values.